server/src/main/java/io/spine/server/delivery/CatchUpStation.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2020, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.delivery;
+
+import com.google.protobuf.Duration;
+import com.google.protobuf.util.Durations;
+import io.spine.core.Event;
+import io.spine.server.delivery.event.CatchUpStarted;
+import io.spine.server.event.EventComparator;
+import io.spine.type.TypeUrl;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.spine.server.delivery.CatchUpStatus.COMPLETED;
+import static io.spine.server.delivery.CatchUpStatus.FINALIZING;
+import static io.spine.server.delivery.CatchUpStatus.IN_PROGRESS;
+import static io.spine.server.delivery.InboxMessageStatus.TO_CATCH_UP;
+import static io.spine.server.delivery.InboxMessageStatus.TO_DELIVER;
+
+/**
+ * A station that performs the delivery of messages to the catching-up targets.
+ *
+ * <h1>Overview</h1>
+ *
+ * <p>Matches the messages on the passed {@link Conveyor} to the known {@link CatchUp} jobs basing
+ * on the target entity type and the identifier of the entity to which the message is sent.
+ *
+ * <p>Depending on the status of the job, the matched messages are processed accordingly. See
+ * more on that below.
+ *
+ * <b>1. Catch-up {@code IN_PROGRESS}.</b>
```

Comment:
Please have these headers coming in separate paragraphs. Now they come as a part of a proceeding paragraph.